logic/resolvable/Unifier.java

```diff
@@ -236,6 +248,14 @@ public void addLabelType(Identifier.Variable.Label source, Set<Label> allowedTyp
             unifiedRequirements.types(target, allowedTypes);
         }
 
+        void addConstantValueRequirements(Set<ValueConstraint<?>> values,
+                                          Retrievable id, Retrievable unifiedId) {
+            iterate(values).filter(v -> v.predicate().equals(EQ)).forEachRemaining(value -> {
+                unifiedRequirements().predicates(unifiedId, valuePredicate(value));
```

Comment:
Filter out the non-EQ constraints here. 
This was the original behaviour (I believe). Removing the filter would allow other constraints to be unification requirements. Should we just do that in this PR?